app/src/main/java/com/example/habitsmasher/ui/dashboard/HabitListFragment.java

```diff
@@ -121,22 +138,25 @@ public void addNewHabit(Habit habit) {
     }
 
     /**
-     * This method is responsible for adding a new habit to the database
+     * This method is responsible for adding a new habit to the firestore database
      * @param title the habit title
      * @param reason the habit reason
      * @param date the habit date
      */
-    public void addHabitToDatabase(String title, String reason, Date date){
+    public void addHabitToDatabase(String title, String reason, Date date) {
         /**
          * Handling of adding a habit to firebase
          */
-        _db = FirebaseFirestore.getInstance();
         final CollectionReference collectionReference = _db.collection("Habits");
-        HashMap<String, Habit> firebaseData = new HashMap<>();
-        firebaseData.put(title, new Habit(title, reason, date));
+        HashMap<String, Object> habitData = new HashMap<>();
+
+        habitData.put("title", title);
+        habitData.put("reason", reason);
+        habitData.put("date", date);
+
         collectionReference
                 .document(title)
-                .set(firebaseData)
+                .set(habitData)
```

Comment:
These couple changes were necessary as we were inserting unnecessary nesting into the database. Instead of having a single `Habit` document with three fields (title, reason, and date), we were creating another habit within the outer habit, which proved to be a problem when fetching.